crates/bevy_ecs/src/component.rs

```diff
@@ -1344,11 +1665,30 @@ impl RequiredComponentConstructor {
     }
 }
 
+/// Metadata associated with a required component. See [`Component`] for details.
+#[derive(Clone)]
+pub struct RequiredComponent {
+    /// The constructor used for the required component.
+    pub constructor: RequiredComponentConstructor,
+
+    /// The depth of the component requirement in the requirement hierarchy for this component.
+    /// This is used for determining which constructor is used in cases where there are duplicate requires.
+    ///
+    /// For example, consider the inheritance tree `X -> Y -> Z`, where `->` indicates a requirement.
+    /// `X -> Y` and `Y -> Z` are direct requirements with a depth of 0, while `Z` is only indirectly
+    /// required for `X` with a depth of `1`.
+    ///
+    /// In cases where there are multiple conflicting requirements with the same depth, a higher priority
+    /// will be given to components listed earlier in the `require` attribute, or to the latest added requirement
+    /// if registered at runtime.
+    pub inheritance_depth: u16,
```

Comment:
This could maybe be a `u8`, since I doubt inheritance trees would exceed a depth of 255, but I went with `u16` to be safe